ddl/ddl_api.go

```diff
@@ -4469,6 +4469,12 @@ func (d *ddl) getModifiableColumnJob(ctx context.Context, sctx sessionctx.Contex
 		CtxVars: []interface{}{needChangeColData},
 		Args:    []interface{}{&newCol, originalColName, spec.Position, modifyColumnTp, newAutoRandBits},
 	}
+	if info := sctx.GetSessionVars().StmtCtx.MultiSchemaInfo; info != nil {
+		info.AddColumns = append(info.AddColumns, newCol.ColumnInfo)
+		if newCol.Name.L != originalColName.L {
+			info.DropColumns = append(info.DropColumns, col.ColumnInfo)
+		}
+	}
```

Comment:
`info.AddColumns` is appended even if it is rename job?

Reply:
I think there is a better way to avoid this, rewrite the check functions now.